### DISCUSSION TITLE
QZ reports double calories

### DISCUSSION THREAD

**Discussion by brokkolo** [cagnulein/qdomyos-zwift#1388] [Q&A] [answered]
my setup is:
- Toorx srx 500 bike + garmin HR chest strap
- samsung s8 running QZ (connected to bike and hrm), luckily i had this old phone laying around as with android10 ant+ works perfect
- garmin forerunner 955 connected to hrm and QZ (ant+ powr & spd/cad)

all the data seems consistant between what is displayed on qz tiles and what garmin watch receives, except for calories! which i believe qz and garmin calculate independently.
Interestingly, qz shows DOUBLE the amount of what garmin calculates, and garmin seems to be the one with the right number.
this is also reported in the Fit file exported by qz.

is this a known issue? how do i tune qz to report the correct number of calories?

**Comment by cagnulein** [comment]
@rbbarbieri attach the debug log from a session and I will check it

**Reply by brokkolo** [reply]
today I got more similar calories results between QZ and Garmin but there was a discrepancy in distance. I'm experimenting with different settings, as soon as i have consistent differences i will get back to you, don't want to waste your time ;)

**Comment by cagnulein** [comment]
ok thanks, no problem!

Il giorno dom 2 apr 2023 alle 17:20 rbbarbieri ***@***.***>
ha scritto:

> today I got more similar calories results between QZ and Garmin but there
> was a discrepancy in distance. I'm experimenting with different settings,
> as soon as i have consistent differences i will get back to you, don't want
> to waste your time ;)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cagnulein/qdomyos-zwift/discussions/1388#discussioncomment-5502255>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALYWDOZMID56ASTIOS5NDW7GKMBANCNFSM6AAAAAAWPQX754>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>
-- 
Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud

**Comment by brokkolo** [comment]
i'm back, did a few rides these days and didn't have the double calories issues again.
but:
my garmin fit file from watch has some discrepancies with the fit file from qz
 - distance 22.1km vs 19.8
 - calories 555 vs 458

I understand that difference in calories is most probably due to difference in settings used for the calculation (garmin might have different power/hr zones or different max hr set)

what i don't get is the difference in distance, as the watch gets it from qz via ant+ and watch and qz both show same cadence and power as I pedal. (i've set the wheel size to 1 00 mm on the watch, could this be it?)
[debug-Thu_Apr_6_10_48_50_2023.log.zip](https://github.com/cagnulein/qdomyos-zwift/files/11170306/debug-Thu_Apr_6_10_48_50_2023.log.zip)
[garmin & qz fit files.zip](https://github.com/cagnulein/qdomyos-zwift/files/11170287/garmin.qz.fit.files.zip)

**Reply by cagnulein** [reply]
the issue is for sure on the wheel size @rbbarbieri you have to experiment about the right ratio. I don't know the current ratio of your bike but you can easily calculate it dividing speed by cadence 

also you can do 22.1/19.8 = 1,116161616161616 so probably set the wheel size to 111 or 89

let me know

**Reply by brokkolo** [reply]
you're super quick in responding!!
ok, will change the wheel parameter and get back

**Reply by brokkolo** [reply, marked as answer]
this is solved for me, changing wheel size made qz and garmin report close to same distances. Calories are still a little off but due to different way of calculating them
thank you
